libfdcore/peers.c

```diff
@@ -197,10 +200,16 @@ int fd_peer_remove ( DiamId_t diamid, size_t diamidlen )
 			peer->p_hdr.info.config.pic_flags.exp = PI_EXP_INACTIVE;
 			peer->p_hdr.info.config.pic_lft = 0;
 			CHECK_FCT( fd_p_expi_update(peer) );
+			found = 1;
 			break;
 		}
 	}
 
+	if (!found)
+	{
+		TRACE_ERROR("Diameter peer %*s not valid - caller is out of sync", (int)diamidlen, diamid);
```

Comment:
Discussed with Andy offline.  The conclusion we came to was that the concern about making the log when its not needed thanks to deletes crossing over could also be made about https://github.com/Metaswitch/cpp-common/blob/master/src/diameterstack.cpp#L338. We decided that the benefit from having some indication at production log level that something might have gone wrong in the synchronization of RealmManager, DiameterStack and freeDiameter peer state was more important than the risk that we might conceivably also get this log thanks to very occasional timing windows in peer shutdown.